test/jdk/javax/sound/sampled/Clip/AudioContentHandlers.java

```diff
@@ -0,0 +1,112 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.sound.SoundClip;
+import javax.sound.sampled.AudioFileFormat;
+import javax.sound.sampled.AudioFormat;
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+
+import static javax.sound.sampled.AudioFileFormat.Type.AIFC;
+import static javax.sound.sampled.AudioFileFormat.Type.AIFF;
+import static javax.sound.sampled.AudioFileFormat.Type.AU;
+import static javax.sound.sampled.AudioFileFormat.Type.SND;
+import static javax.sound.sampled.AudioFileFormat.Type.WAVE;
+
+/**
+ * @test
+ * @bug 8204454 8359053
+ * @summary URL.getContent() should return SoundClip for supported formats
+ * @run main/othervm -Xmx128m AudioContentHandlers
+ */
+public final class AudioContentHandlers {
+
+    private static final List<AudioFormat> formats = new ArrayList<>();
+
+    private static final AudioFormat.Encoding[] encodings =
+            {AudioFormat.Encoding.ALAW, AudioFormat.Encoding.ULAW,
+                    AudioFormat.Encoding.PCM_SIGNED,
+                    AudioFormat.Encoding.PCM_UNSIGNED,
+                    AudioFormat.Encoding.PCM_FLOAT};
+
+    private static final AudioFileFormat.Type[] types =
+            {WAVE, AU, AIFF, AIFC, SND};
+
+    static {
+        for (final AudioFormat.Encoding enc : encodings) {
+            formats.add(new AudioFormat(enc, 44100, 8, 1, 1, 44100, true));
+            formats.add(new AudioFormat(enc, 44100, 8, 1, 1, 44100, false));
+        }
+    }
+
+    public static void main(final String[] args) throws Exception {
+        for (final AudioFileFormat.Type type : types) {
+            for (final AudioFormat format : formats) {
+                File file = new File("audio." + type.getExtension());
+                try {
+                    AudioSystem.write(getStream(format), type, file);
+                } catch (IOException | IllegalArgumentException ignored) {
+                    continue;
+                }
+                SoundClip content;
+                try {
+                    content = (SoundClip) file.toURL().getContent();
+                    // We need to generate OOM because the stream in SoundClip
+                    // will be closed in finalize().
+                    generateOOME();
+                } finally {
+                    Files.delete(file.toPath());
```

Comment:
```suggestion
                    file.delete();
```
Other parts of the test don't use NIO.